### PR TITLE
[MM-42201] Adding error for username conflict on groups

### PR DESCRIPTION
**Pull request by BenCookie95**
#### Summary
Adding a descriptive error for group names matching a username. 

Note: Unable to add test to `update_user_group_modal` due to some issues with functional components and accessing hooks. Will have to dive deeper another time.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-42201

#### Related Pull Requests
https://github.com/mattermost/mattermost-server/pull/19660

#### Screenshots
Screen Shot 2022-03-02 at 11 40 08 AM

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by mkraft — APPROVED** [review]

File: components/update_user_group_modal/update_user_group_modal.tsx [hunk 1]
```diff
@@ -108,6 +108,9 @@ const UpdateUserGroupModal = (props: Props) => {
             if (data.error?.server_error_id === 'app.custom_group.unique_name') {
                 setMentionInputErrorText(Utils.localizeMessage('user_groups_modal.mentionNotUnique', 'Mention needs to be unique.'));
                 setSaving(false);
+            } else if (data.error?.server_error_id === 'app.group.username_conflict') {
+                setMentionInputErrorText(Utils.localizeMessage('user_groups_modal.mentionUsernameConflict', 'A username already exists with this name. Mention must be unique.'));
+                setSaving(false);
```
Comment (mylonsuren) [inline components/update_user_group_modal/update_user_group_modal.tsx:113]: 1/5, Might be worth looking into extracting this repeated logic into a custom hook to handle input errors in the future.

**Review by mylonsuren — APPROVED** [review]
👍

**Review by furqanmlk — APPROVED** [review]
Working as expected

image

**Pull request merged**